src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/data/UnixFilePermissions.xml

```diff
@@ -0,0 +1,20 @@
+<FileList>
+  <File Path="tools/Darwin/aapt2" Permission="755" />
+  <File Path="tools/Darwin/aprofutil" Permission="755" />
+  <File Path="tools/Darwin/illinkanalyzer" Permission="755" />
+  <File Path="tools/Darwin/jit-times" Permission="755" />
+  <File Path="tools/Darwin/mono" Permission="755" />
+  <File Path="tools/Darwin/mono-symbolicate" Permission="755" />
+  <File Path="tools/Darwin/ndk/aarch64-linux-android-as" Permission="755" />
```

Comment:
Maybe it's worth adding target to generate this file so we don't have to worry about declaring the entries twice? We could set the `Permission="foo"` metadata in create-installers and use it to determine what needs to go in this  file:
https://github.com/xamarin/xamarin-android/blob/fc3f02826077dde0caf78e86080261669a0ba44e/build-tools/installers/create-installers.targets#L343

Reply:
I reworked this, so the file is generated now.

I'll need to check the build results to see if the file is correct.

Reply:
This is close, but I think some files are missing:
```xml
<FileList>
  <File Path="tools/Darwin/ndk/aarch64-linux-android-as" Permission="755" />
  <File Path="tools/Darwin/ndk/aarch64-linux-android-ld" Permission="755" />
  <File Path="tools/Darwin/ndk/aarch64-linux-android-strip" Permission="755" />
  <File Path="tools/Darwin/ndk/arm-linux-androideabi-as" Permission="755" />
  <File Path="tools/Darwin/ndk/arm-linux-androideabi-ld" Permission="755" />
  <File Path="tools/Darwin/ndk/arm-linux-androideabi-strip" Permission="755" />
  <File Path="tools/Darwin/ndk/i686-linux-android-as" Permission="755" />
  <File Path="tools/Darwin/ndk/i686-linux-android-ld" Permission="755" />
  <File Path="tools/Darwin/ndk/i686-linux-android-strip" Permission="755" />
  <File Path="tools/Darwin/ndk/x86_64-linux-android-as" Permission="755" />
  <File Path="tools/Darwin/ndk/x86_64-linux-android-ld" Permission="755" />
  <File Path="tools/Darwin/ndk/x86_64-linux-android-strip" Permission="755" />
  <File Path="tools/Darwin/mono" Permission="755" />
  <File Path="tools/Darwin/aapt2" Permission="755" />
</FileList>
```
Why are these files not signed and hardened?
https://github.com/xamarin/xamarin-android/blob/002b5f95b4556809c4083d2dc588b52b4edda6a4/build-tools/installers/create-installers.targets#L355-L357

Reply:
The `_MSBuildFilesUnixSign` and `_MSBuildFilesUnixSignAndHarden` item types are used for Apple notarization.  That system is only concerned with `Mach-O` binary file types:
```
file lib/xamarin.android/xbuild/Xamarin/Android/Darwin/ndk/aarch64-linux-android-as
lib/xamarin.android/xbuild/Xamarin/Android/Darwin/ndk/aarch64-linux-android-as: Mach-O 64-bit executable x86_64 
```
These other files are classified as POSIX shell script text executables:
```
file lib/xamarin.android/xbuild/Xamarin/Android/Darwin/illinkanalyzer
lib/xamarin.android/xbuild/Xamarin/Android/Darwin/illinkanalyzer: POSIX shell script text executable, ASCII text
file lib/xamarin.android/xbuild/Xamarin/Android/Darwin/jit-times
lib/xamarin.android/xbuild/Xamarin/Android/Darwin/jit-times: POSIX shell script text executable, ASCII text
file lib/xamarin.android/xbuild/Xamarin/Android/Darwin/aprofutil
lib/xamarin.android/xbuild/Xamarin/Android/Darwin/aprofutil: POSIX shell script text executable, ASCII text
```

Reply:
Ok, the ones that say `POSIX shell script` are actually shell scripts that just run mono for these tools.

I think what I have is correct now.